src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -93,7 +93,7 @@ internal unsafe SocketError DoOperationConnect(Socket socket, SafeSocketHandle h
             return socketError;
         }
 
-        internal SocketError DoOperationDisconnect(Socket socket, SafeSocketHandle handle)
+        internal SocketError DoOperationDisconnect(Socket socket, SafeSocketHandle handle, CancellationToken cancellationToken)
```

Comment:
Do we check for cancellation anywhere up until this point?  If not, since the token isn't used by the implementation, it might be worth polling the token once here and returning an appropriate cancellation error before proceeding to do the disconnect.

Reply:
There is a test for this -- Disconnect_Precanceled_ThrowsOperationCanceledException -- and it's passing, so we must be checking the cancellation token somewhere in all this.